vmm/src/vm.rs

```diff
@@ -1060,12 +1074,26 @@ impl Vm {
         payload: &PayloadConfig,
         memory_manager: Arc<Mutex<MemoryManager>>,
         #[cfg(feature = "igvm")] cpu_manager: Arc<Mutex<cpu::CpuManager>>,
+        #[cfg(feature = "sev_snp")] sev_snp_enabled: bool,
     ) -> Result<EntryPoint> {
         trace_scoped!("load_payload");
+        #[cfg(feature = "sev_snp")]
+        let host_data = &payload.host_data;
         #[cfg(feature = "igvm")]
-        if let Some(_igvm_file) = &payload.igvm {
-            let igvm = File::open(_igvm_file).map_err(Error::IgvmFile)?;
-            return Self::load_igvm(igvm, memory_manager, cpu_manager);
+        {
+            if let Some(_igvm_file) = &payload.igvm {
+                let igvm = File::open(_igvm_file).map_err(Error::IgvmFile)?;
+                #[cfg(feature = "sev_snp")]
+                if sev_snp_enabled {
+                    if let Some(host_data_str) = host_data {
+                        return Self::load_igvm(igvm, memory_manager, cpu_manager, host_data_str);
+                    } else {
+                        return Self::load_igvm(igvm, memory_manager, cpu_manager, "");
```

Comment:
Have you run any tests with this patch when `host_data` is not provided?  I believe this call will trigger the assertion `assert_eq!(64, host_data.len());` from `igvm::igvm_loader()`. 

You should make the `host_data` as an `Option` to `igvm::igvm_loader`, and handle it accordingly if `host_data` is indeed optional when `sev_snp` is enabled.

Reply:
I tested this patch and it boots fine. Even though host data is empty or not provided, we are passing an array with zeroes. We only update the array if there is something.

Reply:
I see. You had the `assertion` behind the `if !host_data.is_empty()` block.

I still believe your code would be more concise if you make `vmm::vm::Vm::load_igvm()` and `vmm::igvm::igvm_loader::load_igvm` take `host_data: Option<String>`.

Reply:
Let me work on this.